MantidQt/CustomInterfaces/src/Reflectometry/ReflSaveTabPresenter.cpp

```diff
@@ -0,0 +1,199 @@
+          wsNames.begin(), wsNames.end(), validNames.begin(),
+          [rgx](std::string s) { return boost::regex_search(s, rgx); });
+    } catch (boost::regex_error &) {
+      m_mainPresenter->giveUserCritical("Error, invalid regular expression\n",
+                                        "Invalid regex");
+    }
+  } else {
+    // Otherwise simply add names where the filter string is found in
+    it = std::copy_if(wsNames.begin(), wsNames.end(), validNames.begin(),
+                      [filter](std::string s) {
+                        return s.find(filter) != std::string::npos;
+                      });
+  }
+
+  validNames.resize(std::distance(validNames.begin(), it));
+  m_view->setWorkspaceList(validNames);
+}
+
+/** Fills the 'List of Logged Parameters' widget with the parameters of the
+* currently selected workspace
+*/
+void ReflSaveTabPresenter::populateParametersList() {
+  m_view->clearParametersList();
+
+  std::string wsName = m_view->getCurrentWorkspaceName();
+  std::vector<std::string> logs;
+  const auto &properties = AnalysisDataService::Instance()
+                               .retrieveWS<MatrixWorkspace>(wsName)
+                               ->run()
+                               .getProperties();
+  for (auto it = properties.begin(); it != properties.end(); it++) {
+    logs.push_back((*it)->name());
+  }
+  m_view->setParametersList(logs);
+}
+
+/** Saves selected workspaces
+*/
+void ReflSaveTabPresenter::saveWorkspaces() {
+  // Check that save directory is valid
+  std::string saveDir = m_view->getSavePath();
+  if (saveDir.empty() || Poco::File(saveDir).isDirectory() == false) {
+    m_mainPresenter->giveUserCritical("Directory specified doesn't exist or "
+                                      "was invalid for your operating system",
+                                      "Invalid directory");
+    return;
+  }
+
+  // Check that at least one workspace has been selected for saving
+  auto wsNames = m_view->getSelectedWorkspaces();
+  if (wsNames.empty()) {
+    m_mainPresenter->giveUserCritical("No workspaces selected. You must select "
+                                      "the workspaces to save.",
+                                      "No workspaces selected");
+  }
+
+  // Create the appropriate save algorithm
+  bool titleCheck = m_view->getTitleCheck();
+  auto selectedParameters = m_view->getSelectedParameters();
+  bool qResolutionCheck = m_view->getQResolutionCheck();
+  std::string prefix = m_view->getPrefix();
+  int formatIndex = m_view->getFileFormatIndex();
+  std::string algName = saveAlgs[formatIndex];
+  std::string extension = saveExts[formatIndex];
+  IAlgorithm_sptr saveAlg = AlgorithmManager::Instance().create(algName);
+
+  for (auto it = wsNames.begin(); it != wsNames.end(); it++) {
+    // Add any additional algorithm-specific properties and execute
+    if (algName != "SaveANSTOAscii") {
+      if (titleCheck)
+        saveAlg->setProperty("Title", *it);
```

Comment:
Performing functional testing with dataset `argus0026287`, I noticed that when checking option "Title" the new tab prints the name of the workspace in the output file, whereas the old one prints the title, which is `GaAs` in this specific case of `argus000026287`. To replicate the behaviour in the old interface we should print the title instead of the workspace name.